api/src/test/java/com/spotify/ffwd/util/HighFrequencyDetectorTest.java

```diff
@@ -0,0 +1,182 @@
+/*-
+ * -\-\-
+ * FastForward API
+ * --
+ * Copyright (C) 2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.ffwd.util;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.name.Names;
+import com.spotify.ffwd.model.Metric;
+import com.spotify.ffwd.statistics.HighFrequencyDetectorStatistics;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HighFrequencyDetectorTest {
+
+  private final boolean dropHighFrequencyMetric = true;
+  private final int minFrequencyMillisAllowed = 1000;
+  private final long highFrequencyDataRecycleMS = 300_000;
+  private final int minNumberOfTriggers = 2;
+
+  private HighFrequencyDetector detector;
+  private HighFrequencyDetector shortSwapDetector;
+  private Logger log;
+
+  @Mock private HighFrequencyDetectorStatistics statistics;
+
+  @Before
+  public void setup() {
+    log = LoggerFactory.getLogger(getClass());
+    detector = createHighFreqDetector(highFrequencyDataRecycleMS);
+    shortSwapDetector = createHighFreqDetector(500L);
+  }
+
+  public HighFrequencyDetector createHighFreqDetector(long swapMS) {
+    final List<Module> modules = Lists.newArrayList();
+
+    modules.add(
+        new AbstractModule() {
+          @Override
+          protected void configure() {
+            bind(Logger.class).toInstance(log);
```

Comment:
it's not clear to me where `bind` is coming from. Could we import it explicitly? Or it could be obvious to everyone but me... :)

Reply:
oh, it's Guice?

Reply:
yep, and it is never imported explicitly anywhere